include/rwkv/cuda/rwkv.cu

```diff
@@ -619,84 +620,73 @@ unsigned long long getSize(unsigned long long i, unsigned long long n_layers, un
 char* getName(unsigned long long i);
 // ptrs, n_layers, n_embed
 
-
+template <typename DType>
+static inline DType *load_tensor(const DType *input, int64_t size)
+{
+    DType first = ((DType *)input)[0];
+    DType last = ((DType *)input)[size / sizeof(DType) - 1];
+    std::cout << +first << '\t' << +last <<  '\t' << int(size/sizeof(DType)) << '\t' << float(size/1024/1024) << "MB" << std::endl;
+    DType *cuda_mem;
+    cudaMalloc(&cuda_mem, size);
+    cudaMemcpy(cuda_mem, input, size, cudaMemcpyHostToDevice);
+
+    return cuda_mem;
+}
 
 std::tuple<unsigned long long,unsigned long long> load (const std::string& filename, int** ptrs) {
-    std::ifstream binfile(filename, std::ios::in | std::ios::binary);
-    if (!binfile.is_open()) {
-        std::cout << "Error opening file " << filename << std::endl;
-        exit(1);
-    }
+    llama_file file(filename.c_str(), "rb");
+    llama_mmap map(&file);
 
     // get n_layers
     // get n_embed
-    unsigned long long n_layers, n_embed;
-    binfile.read((char*)&n_layers, sizeof(unsigned long long));
-    binfile.read((char*)&n_embed, sizeof(unsigned long long));
+    int64_t n_layers, n_embed;
+    n_layers = *(int64_t*)map.addr;
+    n_embed = *(int64_t*)((char*)map.addr + sizeof(int64_t));
+
     // print
     std::cout << "n_layers: " << n_layers << std::endl;
     std::cout << "n_embed: " << n_embed << std::endl;
 
     
   
+    char* current = (char*)map.addr + sizeof(int64_t) * 2;
+
+    for(int64_t i = 0; i < 46; i++) {
+        char* buffer;
+        int64_t size = getSize(i, n_layers, n_embed);
+
+        if (i == EMBED)
+            continue; // keep EMB on CPU
```

Comment:
will you need to set ptrs[EMBED] = something here?

Reply:
Also, EMBED is undefined in this context. Perhaps the enum needs to be moved to a separate enums.h file?

Reply:
You are right. I will fix it.